### ISSUE TITLE
Typechecker fails to flag changes to fields in a frozen dataclass

### ISSUE THREAD

**Issue by BrutalSimplicity** [microsoft/pyright#1215]
**Describe the bug**
I'm using the Pylance extension in VSCode, and when marking a dataclass as frozen, the typechecker doesn't flag modifications to fields as an error. I am not sure that this is an error, or expected behavior due to internal modifications that dataclass applies that Pyright is unable to type-check.

**To Reproduce**

```python
@dataclasses.dataclass(frozen=True)
class BaseEntity:
    A: str

@dataclasses.dataclass(frozen=True)
class Entity(BaseEntity):
    B: int

e = Entity('1', 2)
e.A = '5'.  # << Will only produce an error at runtime
```

**Expected behavior**

I would expect the assignment to a read-only field to be flagged as an error to prevent mistakes that only show up at runtime.

**VS Code extension**

#immutable #read-only

**Comment by erictraut** [comment]
Enforcement of frozen dataclass types is now implemented. This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.93, which I just published. It will also be included in the next release of Pylance.
